### PR TITLE
add plugin FFI for privacy profiles

### LINKED ISSUE #2000: FFI to create PrivacyProfile from Python

### PR DESCRIPTION
**Pull request by Shoeboxam**
Close #2000

### CODE REVIEW

**Comment by Shoeboxam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/opendp/opendp/2001?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2001** <a href="https://app.graphite.dev/github/pr/opendp/opendp/2001?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1831** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1831?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1860](https://github.com/opendp/opendp/pull/1860) <a href="https://app.graphite.dev/github/pr/opendp/opendp/1860?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1999** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1999?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1768** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1768?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#1854](https://github.com/opendp/opendp/pull/1854) <a href="https://app.graphite.dev/github/pr/opendp/opendp/1854?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#1998** <a href="https://app.graphite.dev/github/pr/opendp/opendp/1998?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Shoeboxam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python/src/opendp/mod.py [hunk 1]
```diff
@@ -631,13 +631,21 @@ def __iter__(self):
...
+        from opendp._data import privacy_profile_delta
+        return privacy_profile_delta(self.curve, epsilon)
+
     def epsilon(self, delta):
-        from opendp._data import smd_curve_epsilon
-        return smd_curve_epsilon(self.curve, delta)
+        from opendp._data import privacy_profile_epsilon
+        return privacy_profile_epsilon(self.curve, delta)
+    
+    def _depends_on(self, *args):
```
Comment (mccalluc) [inline python/src/opendp/mod.py:646]: Seems to be unused?

**Review by mccalluc — COMMENTED** [review]
Need to head out, but I haven't read this to the bottom. High level concern, if the issue being fixed is "PrivacyProfile from Python", should there be demonstration and docs for that here?

File: rust/src/measures/ffi.rs [hunk 2]
```diff
@@ -340,3 +341,30 @@ impl<Q> Default for TypedMeasure<Q> {
 impl<Q> Measure for TypedMeasure<Q> {
     type Distance = Q;
 }
+
+#[bootstrap(
+    name = "new_privacy_profile",
+    features("contrib"),
```
Comment (mccalluc) [inline rust/src/measures/ffi.rs:347]: Add `honest-but-curious`? I feel like that's been standard for user plugins.

**Review by mccalluc — COMMENTED** [review]
One more question.

File: rust/src/measures/ffi.rs (hunk 2 above)
Comment (Shoeboxam) [reply rust/src/measures/ffi.rs:347]: good catch

File: python/src/opendp/mod.py (hunk 1 above)
Comment (mccalluc) [reply python/src/opendp/mod.py:646]: never mind: Found the reference in the codegen.

**Review by mccalluc — COMMENTED** [review]
Thanks for the updates. I still feel that some example of what this does would be useful: Either a doctest that could be included in the API reference, or something in the User Guide? This might be further up the stack, or there might already be an issue, though I'm not seeing it. A new issue would be fine if you don't want to hold up this PR. Sorry if I'm missing something.

**Review by mccalluc — APPROVED** [review]
Thanks!

**Pull request merged**